File: an2_sem4/MPP/Homework/sapt14/network/src/main/java/ro/ubbcluj/cs/stefnmubb/festivalsellpoint/network/dto/SpectacolDTO.java
package ro.ubbcluj.cs.stefnmubb.festivalsellpoint.network.dto;

import ro.ubbcluj.cs.stefnmubb.festivalsellpoint.domain.Entity;
import ro.ubbcluj.cs.stefnmubb.festivalsellpoint.domain.Spectacol;
import ro.ubbcluj.cs.stefnmubb.festivalsellpoint.network.Constants;

import java.time.LocalDateTime;

public class SpectacolDTO extends EntityDTO {
    private final String artist;
    private String data;
    private final String locatie;
    private final int nrLocuriDisponibile;
    private final int nrLocuriVandute;

    public LocalDateTime getData() {
        return LocalDateTime.parse(data, Constants.DTO_FORMATTER);
    }

    public void setData(LocalDateTime data) {
        this.data = data.format(Constants.DTO_FORMATTER);
    }

    public SpectacolDTO(String artist, LocalDateTime data, String locatie, int nrLocuriDisponibile, int nrLocuriVandute) {
        this.artist = artist;
        this.data = data.format(Constants.DTO_FORMATTER);
        this.locatie = locatie;
        this.nrLocuriDisponibile = nrLocuriDisponibile;
        this.nrLocuriVandute = nrLocuriVandute;
    }

    public String getArtist() {
        return artist;
    }

    public String getLocatie() {
        return locatie;
    }

    public int getNrLocuriDisponibile() {
        return nrLocuriDisponibile;
    }

    public int getNrLocuriVandute() {
        return nrLocuriVandute;
    }

    public static SpectacolDTO fromSpectacol(Spectacol s){
        var sp = new SpectacolDTO(s.getArtist(), s.getData(), s.getLocatie(), s.getNrLocuriDisponibile(), s.getNrLocuriVandute());
        sp.setId(s.getId());
        return sp;
    }

    public Spectacol toSpectacol(){
        var sp=new Spectacol(getArtist(), getData(), getLocatie(), getNrLocuriDisponibile(), getNrLocuriVandute());
        sp.setId(getId());
        return sp;
    }
}

File: an2_sem4/MPP/Homework/sapt9/cs/FestivalSellpoint/FestivalSellpoint.Network/DTO/SpectacolDTO.cs
using FestivalSellpoint.Domain;
using FestivalSellpoint.Network.Utils;
using System;

namespace FestivalSellpoint.Network.DTO
{
    [Serializable]
    public class SpectacolDTO : EntityDTO
    {
        public string Artist { get; }

        private string _Data;

        public DateTime Data
        {
            get => DateTime.ParseExact(_Data, Constants.DTODateTimeFormatter, null);
            private set => _Data = value.ToString(Constants.DTODateTimeFormatter);
        }

        public string Locatie { get; }
        public int NrLocuriDisponibile { get; }
        public int NrLocuriVandute { get; }

        public SpectacolDTO(string artist, DateTime data, string locatie, int nrLocuriDisponibile, int nrLocuriVandute)
        {
            Artist = artist;
            Data = data;
            Locatie = locatie;
            NrLocuriDisponibile = nrLocuriDisponibile;
            NrLocuriVandute = nrLocuriVandute;
            Console.WriteLine(_Data);
        }

        public static SpectacolDTO FromSpectacol(Spectacol s)
            => new SpectacolDTO(s.Artist, s.Data, s.Locatie, s.NrLocuriDisponibile, s.NrLocuriVandute) { Id = s.Id };
        public Spectacol ToSpectacol()
            => new Spectacol(Artist, Data, Locatie, NrLocuriDisponibile, NrLocuriVandute) { Id = Id };
    }
}
